File: student/src/main/scala/Common/Info.scala
package Common

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class Info(
                 userName: String,
                 password: String,
                 number: String,
                 volunteerStatus: String,  // 改为字符串类型
                 floor: String,            // 改为字符串类型
                 sectionNumber: String,    // 改为字符串类型
                 seatNumber: String,       // 改为字符串类型
                 violationCount: String,   // 改为字符串类型
                 volunteerHours: String    // 改为字符串类型
               )

object Info {
  implicit val encoder: Encoder[Info] = deriveEncoder[Info]
  implicit val decoder: Decoder[Info] = deriveDecoder[Info]
}

File: login-panel/src/Plugins/CommonUtils/Info.ts
export class BasicInfo {
    userName: string;
    password:string;
    number: string;

    constructor(name: string, password: string,number: string) {
        this.userName = name;
        this.number = number;
        this.password = password;
    }
}
export class Info{
    valid: boolean|undefined;
    userName: string|null;
    number: string|null;
    volunteerStatus: string|null;
    floor: string|null;
    sectionNumber: string|null;
    seatNumber: string|null;
    violationCount:string|null;
    volunteerHours: string|null;
}
export class LoginInfo {
    name: string;
    password:string;
    constructor(name: string, password: string) {
        this.name = name;
        this.password = password;
    }
}
export class SeatInfo{
    floor: number
    section: number
    seatNumber: number
    constructor(floor:number,section:number,seatNumber:number){
        this.floor=floor
        this.section=section
        this.seatNumber=seatNumber
    }
}